cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaRollerTest.java

```diff
@@ -529,6 +584,29 @@ public void testBrokerInRecoveryState(VertxTestContext testContext) throws Inter
                 asList(2, 3, 4, 1));
     }
 
+    @Test
+    public void testBrokerInRecoveryStateForKRaftController(VertxTestContext testContext) throws InterruptedException {
+        PodOperator podOps = mockPodOps(podId ->
+                (podId == 0) ? failedFuture(new TimeoutException("Timeout")) : succeededFuture()
+        );
+
+        Map<String, Object> recoveryState = new HashMap<>();
+        recoveryState.put("remainingLogsToRecover", 10);
+        recoveryState.put("remainingSegmentsToRecover", 100);
+        BrokerState brokerstate = new BrokerState(2, recoveryState);
+
+        TestingKafkaRoller kafkaRoller = new TestingKafkaRoller(null, null, addKraftPodNames(0, 0, 3),
+                podOps,
+                noException(), null, noException(), noException(), noException(),
+                brokerId -> succeededFuture(true),
+                false, null, false, brokerstate, -1);
+
+        doFailingRollingRestart(testContext, kafkaRoller,
+                List.of(0, 1, 2),
+                KafkaRoller.UnforceableProblem.class, "Pod c-kafka-0 is not ready because the broker is performing log recovery. There are  10 logs and 100 segments left to recover.",
```

Comment:
```suggestion
                KafkaRoller.UnforceableProblem.class, "Pod c-kafka-0 is not ready because the broker is performing log recovery. There are 10 logs and 100 segments left to recover.",
```

Reply:
This extra space needs to be also removed in `KafkaRoller` line 400.